### ISSUE TITLE
[Suggestion] Notawallplan's Mass-Suggestion Thread.

### ISSUE THREAD

**Issue by Notawallplan** [Astavie/ThermalLogistics#5]
Feature "Ports"

- ? Servo Multiplexer: Acts like the Chassis pipe from logistics pipes. Has slots for servos (2-8) with 
corresponding compatible redprints to determine filter and operation.

- ? Logistical Nodeduct: An itemduct that will hijack any non-routed item and send it to the lowest scoring logistical filter (or standard filter), else it would just act like an oddly-shaped impulse itemduct. 

- ? Restrictive and Attractive Nodeducts: Essentially dense and vacuum Logistical Nodeducts.

- ? Logistical Shunter: A Device that acts as a firewall and can route between Logistic Itemduct Networks. Requires 5RF/t from the network to operate, and can accept multiple filter redprints per side.

- ? Electrologistical Coil: Energy storage for a logistics network. Power from this is used for more complex network devices.

- ? Tunneling Servo: Two of these are crafted at a time and can be set codes on both outward and inward sides. Place one in one end and one in the other, make them match codes and it creates a new network between the two (or more), cutting it off from the rest of the network. The tunnel can also be filtered and it's I/O type set on both the outward and inward facing side. (Modes are: IN, OUT, DUPLEX, and CLOSED) (This takes advantage of the little-known fact that you **can** put filters and retrievers on pipes and can operate on the network in that fashion.) 

- [ ] Controllable Requestor: Acts exactly like a requestor, but can also fulfill requests made by a Portable Terminal.

- [ ] Portable Terminal: Exactly what it says on the tin, but performs its operations on either the nearest Controllable Requestor or a specific one. Can also request an item to be dumped at a nodeduct trough manual control.

- [x] Crafting Terminal

New Things

- ? Logistical Shepard: A Machine that can be programmed to do specific tasks on a Logistics network when requested or on an automatic basis. Can also act as a Shunter because of the command set available. Base RF/T is 20RF/t per network connected and its base max is 100RF/t during computation. Base speed is 2 seconds (60ticks) per operation

- ? Relay Shunter: A Device that can change, switch between, and link together Redstone Relay channels.

- ? Packet Spoofer: Can create a signal to the network that will be performed through ghost items. Can be used for network diagnostics.

- ? Logistical Adaptor: Allows logistic actions to act on a different pipe network (ex, allowing logistics over a Buildcraft pipe network.)

Minor enhancements

- [x] JEI recipe autofill for Crafter and Crafting Terminal.

- ? Art on face and back of Thermal Logistics pipe-based devices to help differentiate between them at a glance.

- [x] Terminals being able to be placed on their own and ferry items around like a nodeduct.

- ? Crafters being able to operate mid-pipe for operations on sided blocks.

(Will add more trough edits as I come up with them.)

**Comment by Fortist** [comment]
Wireless Terminal and Crafting Terminal.

**Comment by hron84** [comment]
Minor enhancement:

 - Add a servo slot to the Terminal that enables a button that sends items back to the network from its internal inventory.

**Comment by Astavie** [comment]
You can already click with an item on the terminal to send it back

**Comment by Trinsdar** [comment]
Minor suggestion: add a warning when you don't have the nessacary items or the ability to autocraft the nessacary items for a requested crafting task.

**Comment by Astavie** [comment]
@Trinsdar is that not already a thing?

**Comment by Trinsdar** [comment]
Not exactly. It just shows the cancel x next to the recipe, but no other information afaik

**Comment by SFort** [comment]
Adding either or both of these modes to the network manager would be great:
- clone mode
clones the settings of the pipe
- reset / clear active crafts mode 
would be great so that if for whatever reason you want to make the crafter forget what it was crafting you can without having to replace the crafter and re-inserting the recepie/s it had.

**Comment by Astavie** [comment]
You can already clone crafters with reprint @SFort

**Comment by SFort** [comment]
is that a mode on the manager? (if so I can only see Linking and Monitoring, sorry if i'm missing something obvious)

**Comment by Astavie** [comment]
It's an item from thermal

**Comment by SFort** [comment]
ah a "redprint", thanks
